Detectors/FIT/FT0/workflow/src/ReconstructionSpec.cxx

```diff
@@ -55,11 +58,11 @@ void ReconstructionDPL::run(ProcessingContext& pc)
   if (mUseMC) {
     LOG(info) << "Ignoring MC info";
   }
-  // auto caliboffsets = pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
-  //  mReco.SetChannelOffset(caliboffsets.get());
-  auto caliboffsets = mCCDBManager.get<o2::ft0::FT0ChannelTimeCalibrationObject>("FT0/Calibration/ChannelTimeOffset");
-  mReco.SetChannelOffset(caliboffsets);
-  LOG(debug) << " RecoSpec  mReco.SetChannelOffset(&caliboffsets)";
+  if (mUpdateCCDB) {
+    auto caliboffsets = pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
+    mReco.SetChannelOffset(caliboffsets.get());
+    LOG(info) << "RecoSpec  mReco.SetChannelOffset(&caliboffsets)";
+  }
```

Comment:
This will work since you have just 1 CCDBobject, in general, you don't need to set again the pointer of the calib object here. The idea of using the `finaliseCCDB` method is:
1) here you simply call
```
if (mUpdateCCDB) {
  pc.inputs().get<o2::ft0::FT0ChannelTimeCalibrationObject*>("ft0offsets");
}
```
Note that you ignore the return value: this will check if the object needs to be updated (which is always a case at the 1st call) and if it is, will call the `finaliseCCDB` with the matcher and pointer of this object.
2) in the finaliseCCDB (called for each CCDB object and only when it is updated) you just do
```
if (matcher == ConcreteDataMatcher("FT0", "TimeOffset", 0)) {
 mReco.SetChannelOffset((o2::ft0::FT0ChannelTimeCalibrationObject*)obj)
 LOG(info) << "TimeOffset updated";
  mUpdateCCDB = false; // only if you don't want to check anymore the validity of the object (though it costs nothing) and note that this is dangerous if you have multiple ccdb objects, it is safer to keep this line in the `if (mUpdateCCDB) ...` above.
return;
}
```